auth/src/main/java/org/aerogear/auth/impl/OIDCTokenAuthenticatorImpl.java

```diff
@@ -26,4 +53,68 @@ public Principal authenticate(ICredential credential) throws AuthenticationExcep
         // This authenticator can't manage this type of credential
         return null;
     }
+
+    public void logout(Principal principal) {
+        // Get user's identity token
+        OIDCCredentials credentials = (OIDCCredentials) ((IUserPrincipal) principal).getCredentials();
+        String identityToken = credentials.getIdentityToken();
+
+        // Construct the logout URL
+        URL logoutUrl = parseLogoutURL(identityToken);
+        String serverHostname = logoutUrl.getHost();
+
+        // Build the OkHttpClient client
+        OkHttpClient client = buildHttpClient(logoutUrl, serverHostname);
+
+        // Construct and invoke logout request
+        performLogout(client, logoutUrl);
+    }
+
+    private void performLogout(OkHttpClient client, URL logoutUrl) {
+        HttpServiceModule serviceModule = new OkHttpServiceModule(client);
+
+        HttpRequest request = serviceModule.newRequest();
+        request.get(logoutUrl.toString());
+
+        HttpResponse response = request.execute();
+        response.onComplete(new LogoutCompleteHandler(response, AuthStateManager.getInstance())); // will compile when AuthStateManaer is refactored to singleton
+        response.waitForCompletionAndClose();
+    }
+
+    private OkHttpClient buildHttpClient(URL logoutUrl, String serverHostname) {
+        HttpsURLConnection connection = null;
+        try {
+            connection = (HttpsURLConnection) logoutUrl.openConnection();
+        } catch (IOException e) {
+            e.printStackTrace();
+
+        }
+
+        SSLSocketFactory sslSocketFactory = TrustKit.getInstance().getSSLSocketFactory(serverHostname);
```

Comment:
I don't think we should have trustkit/pinning yet. We can remove any reference to that and these 3 lines here.